### ISSUE TITLE
Decorators should not return undefined when used as Decorator-Factory

### ISSUE THREAD

**Issue by andreasrueedlinger** [lukeautry/tsoa#16]
Since you use your decorators as decorator-factories (https://www.typescriptlang.org/docs/handbook/decorators.html#decorator-factories), the method should return a function itself and not undefined.

An error is thrown if you - for example - include the reflect-metadata package in your project.

You can easily change your decorators to follow the definition:
```typescript
export function Get(value?: string): () => any {
  () => {};
}
```

**Comment by lukeautry** [comment]
@andreasrueedlinger Check out #17 - if that works for you we can merge it.

**Comment by andreasrueedlinger** [comment]
Yes that should work!

**Comment by lukeautry** [comment]
@andreasrueedlinger This has been merged and released in 0.0.61.
